File: Mika/tags/PKG-adhoco-2/core-vm/java/java/sql/Date.java
/**************************************************************************
* Copyright  (c) 2001 by Acunia N.V. All rights reserved.                 *
*                                                                         *
* This software is copyrighted by and is the sole property of Acunia N.V. *
* and its licensors, if any. All rights, title, ownership, or other       *
* interests in the software remain the property of Acunia N.V. and its    *
* licensors, if any.                                                      *
*                                                                         *
* This software may only be used in accordance with the corresponding     *
* license agreement. Any unauthorized use, duplication, transmission,     *
*  distribution or disclosure of this software is expressly forbidden.    *
*                                                                         *
* This Copyright notice may not be removed or modified without prior      *
* written consent of Acunia N.V.                                          *
*                                                                         *
* Acunia N.V. reserves the right to modify this software without notice.  *
*                                                                         *
*   Acunia N.V.                                                           *
*   Vanden Tymplestraat 35      info@acunia.com                           *
*   3000 Leuven                 http://www.acunia.com                     *
*   Belgium - EUROPE                                                      *
**************************************************************************/

/*
** $Id: Date.java,v 1.1.1.1 2004/07/12 14:07:46 cvs Exp $
*/

package java.sql;

import java.io.*;
import java.util.*;

public class Date extends java.util.Date implements Cloneable, Comparable, Serializable {

  private static final long serialVersionUID = 1511598038487230103L;

  public Date(int year, int month, int day) {
    super(year, month, day);
  }
  
  public Date(long date) {
    super(date);
  }

  public int getHours() {
    System.out.println("[java.sql.Date] getHours is deprecated.");
    return 0;
  }
  
  public int getMinutes() {
    System.out.println("[java.sql.Date] getMinutes is deprecated.");
    return 0;
  }
  
  public int getSeconds() {
    System.out.println("[java.sql.Date] getSeconds is deprecated.");
    return 0;
  }
  
  public void setHours(int i) {
    System.out.println("[java.sql.Date] setHours is deprecated.");
  }
  
  public void setMinutes(int i) {
    System.out.println("[java.sql.Date] setMinutes is deprecated.");
  }
  
  public void setSeconds(int i) {
    System.out.println("[java.sql.Date] setSeconds is deprecated.");
  }
  
  public void setTime(long date) {
    super.setTime(date);
  }
  
  public String toString() {
    GregorianCalendar cal = new GregorianCalendar();
    cal.setTime(this);
  
    int year = cal.get(Calendar.YEAR);
    int month = cal.get(Calendar.MONTH) + 1;
    int day = cal.get(Calendar.DAY_OF_MONTH);

    return "" + year + "-" +
                (month < 10 ? "0" + month : "" + month) + "-" +
                (day < 10 ? "0" + day : "" + day);
  }

  public static Date valueOf(String s) {
    StringTokenizer tok = new StringTokenizer(s, "-");
    
    int year = Integer.parseInt(tok.nextToken());
    int month = Integer.parseInt(tok.nextToken());
    int day = Integer.parseInt(tok.nextToken());

    return new Date(year, month, day);
  }

}

File: Mika/tags/PKG-adhoco-2/core-vm/src/native/java/util/Date.c
/**************************************************************************
* Copyright (c) 2001, 2002, 2003 by Acunia N.V. All rights reserved.      *
*                                                                         *
* This software is copyrighted by and is the sole property of Acunia N.V. *
* and its licensors, if any. All rights, title, ownership, or other       *
* interests in the software remain the property of Acunia N.V. and its    *
* licensors, if any.                                                      *
*                                                                         *
* This software may only be used in accordance with the corresponding     *
* license agreement. Any unauthorized use, duplication, transmission,     *
*  distribution or disclosure of this software is expressly forbidden.    *
*                                                                         *
* This Copyright notice may not be removed or modified without prior      *
* written consent of Acunia N.V.                                          *
*                                                                         *
* Acunia N.V. reserves the right to modify this software without notice.  *
*                                                                         *
*   Acunia N.V.                                                           *
*   Philips site 5, box 3       info@acunia.com                           *
*   3001 Leuven                 http://www.acunia.com                     *
*   Belgium - EUROPE                                                      *
*                                                                         *
* Modifications copyright (c) 2004 by Chris Gray, /k/ Embedded Java       *
* Solutions. All rights reserved.                                         *
*                                                                         *
**************************************************************************/

/*
** $Id: Date.c,v 1.3 2004/11/18 23:51:52 cvs Exp $
*/

#include <string.h>

#include "core-classes.h"
#include "clazz.h"
#include "dates.h"
#include "heap.h"
#include "ts-mem.h"
#include "threads.h"
#include "wstrings.h"

#define isLeapYear(y) (!((y)%4)&&(((y)<1582)((y)%100)||!((y)%400)))

void Date_create_ymdhms(JNIEnv *env, w_instance thisDate, w_int year, w_int month, w_int date, w_int hours, w_int minutes, w_int seconds){
  w_date wonkadate;
  w_long ttime;

  wonkadate = allocMem(sizeof(w_Date));
  if (!wonkadate) {
    wabort(ABORT_WONKA, "Unable to allocate wonkadate\n");
  }
  wonkadate->msec = 0;
  wonkadate->second = seconds;
  wonkadate->minute = minutes;
  wonkadate->hour = hours;
  wonkadate->day = date;
  wonkadate->month = month;
  wonkadate->year = year+1900;
  ttime = date2millis(wonkadate);
  releaseMem(wonkadate);

  setLongField(thisDate, F_Date_millisSince1jan1970, ttime);

}

